db/data/20220613134926_backfill_institution_uuid_for_degrees.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class BackfillInstitutionUuidForDegrees < ActiveRecord::Migration[6.1]
+  def up
+    degrees = Degree.where.not(uk_degree: nil)
+
+    degrees.find_each do |degree|
+      next if hesa_code_for_institute(degree) == []
+
+      institution = DfE::ReferenceData::Degrees::INSTITUTIONS.some({ hesa_itt_code: hesa_code_for_institute(degree)&.first })
+      degree.update(institution_uuid: institution&.first&.id)
+    end
+
+    degrees = degrees.where(institution_uuid: nil)
+
+    degrees.find_each do |degree|
+      degree.update(institution_uuid: DfE::ReferenceData::Degrees::INSTITUTIONS.some({ name: degree.institution })&.first&.id)
```

Comment:
I realise we said we could split this off, but it might be quite a quick win and we could get it into the same migration - how about:

```suggestion
      institutions = DfE::ReferenceData::Degrees::INSTITUTIONS.all.filter do |record|
        ([record.name] + record.suggestion_synonyms + record.match_synonyms).include?(degree.institution)
      end
      
      degree.update(institution_uuid: institutions&.first&.id)
```

Reply:
really good idea! I've implemented it this way and we got a few more so that's great. I've updated the trello ticket with the stats